packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -96,26 +79,40 @@ function NavigationMenuSelector( {
 	const menuUnavailable =
 		hasResolvedNavigationMenus && currentMenuId === null;
 
-	useEffect( () => {
-		if ( ! hasResolvedNavigationMenus ) {
-			setSelectorLabel( __( 'Loading …' ) );
-		} else if ( noMenuSelected || noBlockMenus || menuUnavailable ) {
-			setSelectorLabel( __( 'Select menu' ) );
-			setEnableOptions( shouldEnableMenuSelector );
-		}
+	let selectorLabel = '';
 
+	if ( isCreatingMenu || isResolvingNavigationMenus ) {
```

Comment:
Made this loading state conditional on actually being _in_ a resolving state rather than whether we _have_ resolved or not.